src/main/java/com/google/edith/servlets/UserStatsServlet.java

```diff
@@ -0,0 +1,90 @@
+package com.google.edith.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.io.BufferedReader;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.stream.Collectors;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Servlet that returns some example content.
+ */
+@WebServlet("/user-stats-servlet")
+public class UserStatsServlet extends HttpServlet {
+
+  private final DatastoreService datastore;
+  private UserInsightsInterface userInsights;
+
+  public UserStatsServlet() {
+    this.datastore = DatastoreServiceFactory.getDatastoreService();
+    this.userInsights = new UserInsights("userId");
+  }
+
+  public UserStatsServlet(DatastoreService datastore, UserInsightsInterface userInsights) {
+    this.datastore = datastore;
+    this.userInsights = userInsights;
+  } 
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/json");   
+    response.getWriter().println(userInsights.createJson());
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    StringBuilder sb = new StringBuilder();
```

Comment:
use stringBuilder as variable name instead of abbreviation

Reply:
Okay!